app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,63 @@
+class OrdersController < ApplicationController
+  before_action :require_admin, only: [ :index, :toggle_status ]
+  before_action :set_order, only: [ :toggle_status, :destroy, :cancel ]
+
+  def index
+    @orders = Order.includes(:user, :order_items, :products).order(created_at: :desc)
+  end
+
+  def my_orders
+    @orders = current_user.orders.includes(:order_items, :products).order(created_at: :desc)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+    unless @order.user == current_user || current_user&.admin?
+      redirect_to my_orders_path, alert: "You can only view your own orders."
+    end
```

Comment:
You should have a relation between users and orders so you can search it like this: `current_user.orders.find(params[:id])`